app/modules/encounters/models.py

```diff
@@ -104,12 +104,26 @@ def add_assets_no_context(self, asset_list):
         for asset in asset_list:
             self.add_asset_no_context(asset)
 
+    # we dont delete .assets here because that is complex (due to annotations)
     def delete(self):
         with db.session.begin():
             while self.assets:
+                # this is actually removing the EncounterAssets joining object (not the assets)
                 db.session.delete(self.assets.pop())
             db.session.delete(self)
 
+    def delete_cascade(self):
+        assets = self.get_assets()
+        # TODO modify behavior when we have annotations
+        with db.session.begin(subtransactions=True):
+            while self.assets:
+                # this is actually removing the EncounterAssets joining object (not the assets)
+                db.session.delete(self.assets.pop())
+            db.session.delete(self)
```

Comment:
Is it safe/sane to delete self and then iterate through the assets owned by self?

Reply:
i think its fine, cuz technically there is no constraint of any kind on those assets _needing_ that encounter that references them.

this code is going to be modified once we have annots in place anyway, as we will be selectively choosing _which_ assets to delete depending on if they are "used" elsewhere.  (i.e. have annotations in use by _other encounters_.)   so i am pretty sure this is good for now in terms of cleanup.